variable/test/transform_test.cpp

```diff
@@ -161,6 +161,28 @@ TEST_F(TransformUnaryTest, slice) {
   }
 }
 
+TEST_F(TransformUnaryTest, transpose) {
```

Comment:
Is this a test that avoids the special cases, as we discussed? If so mention in a comment why this is an important test.

Furthermore, maybe we need to have a better test avoiding the special cases? For a transpose of a unary someone might eventually add an optimization to transpose the input in `transform` before the iteration starts, which would bring us back to the special stride optimization...

Can you instead test for a slice of a higher-dimensional variable?

Reply:
> Is this a test that avoids the special cases, as we discussed? If so mention in a comment why this is an important test.
> 
> Furthermore, maybe we need to have a better test avoiding the special cases? For a transpose of a unary someone might eventually add an optimization to transpose the input in `transform` before the iteration starts, which would bring us back to the special stride optimization...
> 
> Can you instead test for a slice of a higher-dimensional variable?

Old comments not addressed (yet)?

Reply:
Doesn't `TransformUnaryTest.slice` do this? I added some more slices and it should now test the fallback case.

Reply:
Probably yes. But can you nevertheless add a comment explaining the importance of certain tests?